src/SDK/Subscriber/Dispatcher.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Subscriber;
+
+class Dispatcher
+{
+    private static array $Listener = [];
+    private static $instance = null;
+
+    private function __construct()
+    {
+    }
+
+    public static function getInstance()
+    {
+        if (self::$instance == null) {
+            self::$instance = new Dispatcher();
+        }
+
+        return self::$instance;
+    }
+
+    public function setListener(string $event, string $listener):void
+    {
+        self::$Listener[$event] = $listener;
```

Comment:
do we want to allow for multiple listeners for an event?

Reply:
That's would be expected bahavior for event listeners, however technically you can chain a Listener, which in turn could have multiple Subscribers and fan out like that.